scene/resources/particle_process_material.cpp

```diff
@@ -987,6 +983,10 @@ void ParticleProcessMaterial::_update_shader() {
 		// The following snippet causes massive performance hit. We don't need it as long as collision is disabled.
 		// Refer to GH-83744 for more info.
 		if (collision_mode != COLLISION_DISABLED) {
+			// Set the noise direction to up so the turbulence color is "neutral".
+			code += "		if (COLLIDED && length(VELOCITY) <= 3.0) {\n";
+			code += "			noise_direction = vec3(1.0, 0.0, 0.0);\n";
+			code += "		}\n";
```

Comment:
Comparing this to the code in 4.1 I am noticing a few issues worth thinking about. 
1: In 4.1 this equivalent branch did nothing as ``noise_direction`` was never used again:
https://github.com/godotengine/godot/blob/399c9dc393f6f84c0b4e4d4117906c70c048ecf2/scene/resources/particle_process_material.cpp#L873 CC @RPicster Do you remember the original reason for this code?
2. ``noise_direction`` is only ever used if ``!COLLIDED`` is true which means that this code will never impact anything anyway.
3. (This is a pre-existing issue) When collisions are disabled, the old code still ran the calculation for final velocity: https://github.com/godotengine/godot/blob/399c9dc393f6f84c0b4e4d4117906c70c048ecf2/scene/resources/particle_process_material.cpp#L720-L734

These two issues make me suspect that the original code setting ``noise_direction = vec3(1.0, 0.0, 0.0)`` was actually out of date before it was needed and can actually be removed entirely instead of moved here. 

I suggest that while removing that code, you also change the logic to match that in 4.1 so the ``!COLLIDED`` check is only added when using RIGID BODY collisions.

Reply:
Yes you are right, `noise_direction = vec3(1.0, 0.0, 0.0);\n"; ` is indeed useless now.